graphql/errcode/codes.go

```diff
@@ -0,0 +1,49 @@
+package errcode
+
+import (
+	"github.com/vektah/gqlparser/gqlerror"
+)
+
+const ValidationFailed = "GRAPHQL_VALIDATION_FAILED"
+const ParseFailed = "GRAPHQL_PARSE_FAILED"
+
+type ErrorKind int
+
+const (
+	// issues with graphql (validation, parsing).  422s in http, GQL_ERROR in websocket
+	KindProtocol ErrorKind = iota
+	// user errors, 200s in http, GQL_DATA in websocket
+	KindUser
+)
+
+var codeType = map[string]ErrorKind{
+	ValidationFailed: KindProtocol,
+	ParseFailed:      KindProtocol,
+}
+
+// RegisterErrorType should be called by extensions that want to customize the http status codes for errors they return
+func RegisterErrorType(code string, kind ErrorKind) {
+	codeType[code] = kind
+}
+
+// Set the error code on a given graphql error extension
+func Set(err *gqlerror.Error, value string) {
+	if err.Extensions == nil {
+		err.Extensions = map[string]interface{}{}
+	}
+
+	err.Extensions["code"] = value
```

Comment:
I think `"code"` is little bit commonly used in user namespace.
shoud we have an API to change property name?

Reply:
It matches the purpose and name of the apollo [code](https://www.apollographql.com/docs/apollo-server/data/errors/#codes) so I dont think we should have any issues here.